packages/warehouses/src/warehouseClients/DatabricksWarehouseClient.ts

```diff
@@ -114,67 +96,109 @@ enum DatabricksDataTypes {
     TINY_INT = -6,
 }
 
-const convertDataTypeToDimensionType = (type: number): DimensionType => {
+const mapFieldType = (type: DatabricksFieldTypes): DimensionType => {
     switch (type) {
-        case DatabricksDataTypes.BIT:
-        case DatabricksDataTypes.BINARY:
+        case DatabricksFieldTypes.BIT:
+        case DatabricksFieldTypes.BINARY:
             return DimensionType.BOOLEAN;
-        case DatabricksDataTypes.TINY_INT:
-        case DatabricksDataTypes.SMALL_INT:
-        case DatabricksDataTypes.INTEGER:
-        case DatabricksDataTypes.BIG_INT:
-        case DatabricksDataTypes.FLOAT:
-        case DatabricksDataTypes.REAL:
-        case DatabricksDataTypes.DOUBLE_PRECISION:
-        case DatabricksDataTypes.DECIMAL:
-        case DatabricksDataTypes.NUMERIC:
+        case DatabricksFieldTypes.TINY_INT:
+        case DatabricksFieldTypes.SMALL_INT:
+        case DatabricksFieldTypes.INTEGER:
+        case DatabricksFieldTypes.BIG_INT:
+        case DatabricksFieldTypes.FLOAT:
+        case DatabricksFieldTypes.REAL:
+        case DatabricksFieldTypes.DOUBLE_PRECISION:
+        case DatabricksFieldTypes.DECIMAL:
+        case DatabricksFieldTypes.NUMERIC:
             return DimensionType.NUMBER;
-        case DatabricksDataTypes.DATE:
+        case DatabricksFieldTypes.DATE:
             return DimensionType.DATE;
-        case DatabricksDataTypes.TIMESTAMP:
+        case DatabricksFieldTypes.TIMESTAMP:
             return DimensionType.TIMESTAMP;
         default:
             return DimensionType.STRING;
     }
 };
 
 export class DatabricksWarehouseClient implements WarehouseClient {
-    connectionString: string;
+    schema: string;
+
+    catalog: string;
+
+    connectionOptions: ConnectionOptions;
 
     constructor({
         serverHostName,
         port,
         personalAccessToken,
         httpPath,
+        database,
+        dbname = 'SPARK',
     }: CreateDatabricksCredentials) {
-        this.connectionString = `Driver=${DRIVER_PATH};Server=${serverHostName};HOST=${serverHostName};PORT=${port};SparkServerType=3;Schema=default;ThriftTransport=2;SSL=1;AuthMech=3;UID=token;PWD=${personalAccessToken};HTTPPath=${httpPath};UseNativeQuery=1`;
+        this.schema = database;
+        this.catalog = dbname;
```

Comment:
Are we using `this.catalog` or `dbname` anywhere?

Reply:
yes, when we open a session
```
            session = await connection.openSession({
                initialCatalog: this.catalog,
                initialSchema: this.schema,
            });
```